src/Build.UnitTests/BackEnd/GenerateTemporaryTargetAssembly_Tests.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Microsoft.Build.UnitTests;
+using Shouldly;
+using Xunit;
+
+namespace Microsoft.Build.Engine.UnitTests.BackEnd
+{
+    sealed public class GenerateTemporaryTargetAssembly_Tests
+    {
+        [Fact]
+        public void FailsWithOnlyTargetErrors()
+        {
+            MockLogger logger = ObjectModelHelpers.BuildProjectExpectFailure(@"
+<Project>
+    <UsingTask TaskName=""FailingBuilderTask"" AssemblyName=""Microsoft.Build.Engine.UnitTests"" />
+    <Target Name=""MyTarget"">
+        <FailingBuilderTask CurrentProject="".\otherproj.csproj"" />
```

Comment:
Can you make the other project a real project that has a target with an error, and assert that the sole error is that error?